File: scripts/calculator.cpp
#include <iostream>
#include <cmath>

double add(double a, double b) {
    return a + b;
}

double subtract(double a, double b) {
    return a - b;
}

double multiply(double a, double b) {
    return a * b;
}

double divide(double a, double b) {
    if (b == 0) {
        std::cout << "Error: Division by zero" << std::endl;
        return 0;
    }
    return a / b;
}

int modulus(int a, int b) {
    return a % b;
}

double power(double a, double b) {
    return pow(a, b);
}

double squareRoot(double a) {
    if (a < 0) {
        std::cout << "Error: Negative number" << std::endl;
        return -1;
    }
    return sqrt(a);
}

bool isEven(int a) {
    return a % 2 == 0;
}

bool isOdd(int a) {
    return a % 2 != 0;
}

double calculate(std::string operation, double a, double b) {
    if (operation == "add") {
        return add(a, b);
    } else if (operation == "subtract") {
        return subtract(a, b);
    } else if (operation == "multiply") {
        return multiply(a, b);
    } else if (operation == "divide") {
        return divide(a, b);
    } else if (operation == "modulus") {
        return modulus(a, b);
    } else if (operation == "power") {
        return power(a, b);
    } else if (operation == "sqrt") {
        return squareRoot(a);
    } else {
        std::cout << "Error: Invalid operation" << std::endl;
        return -1;
    }
}

int main() {
    double a = 10, b = 5;
    std::cout << "Addition: " << calculate("add", a, b) << std::endl;
    std::cout << "Subtraction: " << calculate("subtract", a, b) << std::endl;
    std::cout << "Multiplication: " << calculate("multiply", a, b) << std::endl;
    std::cout << "Division: " << calculate("divide", a, b) << std::endl;
    std::cout << "Modulus: " << calculate("modulus", a, b) << std::endl;
    std::cout << "Power: " << calculate("power", 2, 3) << std::endl;
    std::cout << "Square Root: " << calculate("sqrt", 16, 0) << std::endl;
    std::cout << "Is 10 even? " << (isEven(10) ? "Yes" : "No") << std::endl;
    std::cout << "Is 5 odd? " << (isOdd(5) ? "Yes" : "No") << std::endl;
    return 0;
}

File: scripts/calculator.lua
local function add(a, b) 
    return a + b 
end

local function subtract(a, b) 
    return a - b 
end

local function multiply(a, b) 
    return a * b 
end

local function divide(a, b) 
    if b == 0 then
        return "Error: Division by zero"
    else
        return a / b
    end
end

local function modulus(a, b) 
    return a % b 
end

local function power(a, b)
    return a ^ b
end

local function squareRoot(a)
    if a < 0 then
        return "Error: Negative number"
    else
        return math.sqrt(a)
    end
end

local function isEven(a)
    return a % 2 == 0
end

local function isOdd(a)
    return a % 2 ~= 0
end

local function calculate(operation, a, b)
    if operation == "add" then
        return add(a, b)
    elseif operation == "subtract" then
        return subtract(a, b)
    elseif operation == "multiply" then
        return multiply(a, b)
    elseif operation == "divide" then
        return divide(a, b)
    elseif operation == "modulus" then
        return modulus(a, b)
    elseif operation == "power" then
        return power(a, b)
    elseif operation == "sqrt" then
        return squareRoot(a)
    else
        return "Error: Invalid operation"
    end
end

print("Addition: " .. calculate("add", 10, 5))
print("Subtraction: " .. calculate("subtract", 10, 5))
print("Multiplication: " .. calculate("multiply", 10, 5))
print("Division: " .. calculate("divide", 10, 5))
print("Modulus: " .. calculate("modulus", 10, 5))
print("Power: " .. calculate("power", 2, 3))
print("Square Root: " .. calculate("sqrt", 16, 0))
print("Is 10 even? " .. (isEven(10) and "Yes" or "No"))
print("Is 5 odd? " .. (isOdd(5) and "Yes" or "No"))